server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -535,14 +536,13 @@ protected void setMinSeqNoToKeep(long seqNo) {
     @Override
     protected Releasable drainSync() {
         try {
-            if (syncPermit.tryAcquire(SYNC_PERMIT, 1, TimeUnit.MINUTES)) {
+            if (syncPermit.tryAcquire(timeValueMinutes(1L)) != null) {
                 boolean result = pauseSync.compareAndSet(false, true);
-                assert result && syncPermit.availablePermits() == 0;
+                assert result && syncPermit.isHeldByCurrentThread();
                 logger.info("All inflight remote translog syncs finished and further syncs paused");
                 return Releasables.releaseOnce(() -> {
-                    syncPermit.release(SYNC_PERMIT);
+                    syncPermit.close();
                     boolean wasSyncPaused = pauseSync.getAndSet(false);
-                    assert syncPermit.availablePermits() == SYNC_PERMIT : "Available permits is " + syncPermit.availablePermits();
```

Comment:
Is there some equivalent thing we can assert here?  Such as
```java
assert !syncPermit.isHeldByCurrentThread() : "Lock is still held by the current thread";
```

Reply:
That makes the contextual meaning clearer. 👍